ambassador/ambassador/ir/ir.py

```diff
@@ -440,6 +443,68 @@ def add_mapping(self, aconf: Config, mapping: IRBaseMapping) -> Optional[IRBaseM
         else:
             return None
 
+    def cluster_ingresses_to_mappings(self, aconf):
+        cluster_ingresses = aconf.get_config("ClusterIngress")
+        if cluster_ingresses is None:
+            return
+
+        for ci_name, ci in cluster_ingresses.items():
+            self.logger.debug("Parsing ClusterIngress {}".format(ci_name))
+
+            ci_rules = ci.get('rules', [])
+            for ci_rule in ci_rules:
+                ci_hosts = ci_rule.get('hosts', [])
+
+                mapping_host = ""
+                for ci_host in ci_hosts:
+                    if mapping_host == "":
+                        mapping_host = ci_host
+                    else:
+                        mapping_host += "|" + ci_host
+
+                ci_http = ci_rule.get('http', None)
+                if ci_http is not None:
+                    ci_paths = ci_http.get('paths', [])
+                    for ci_path in ci_paths:
+                        ci_headers = ci_path.get('appendHeaders', [])
+
+                        ci_splits = ci_path.get('splits', [])
+                        for ci_split in ci_splits:
+                            mapping_identifier = ci_name + '-' + ''.join(random.sample(string.ascii_lowercase, 5))
```

Comment:
Hmmmmmm. Does this really _need_ to be nondeterministic? Is there something deterministic we can use instead? There are two reasons I ask:

1. It makes certain classes of transcoding tests much harder, and
2. If the name of the mapping changes on every reconfigure, it'll be much harder for Jane and for Envoy to follow which things are changing and which are not.

Reply:
makes sense, fixed.